tests/net_hosted/uasyncio_start_server.py

```diff
@@ -25,6 +24,44 @@ async def test():
         print("sleep")
         await asyncio.sleep(0)
 
+    # Test that cancellation works before the server starts if
+    # the subsequent code raises.
+    print("create server3")
+    server3 = await asyncio.start_server(None, "0.0.0.0", 8000)
+    try:
+        async with server3:
+            raise OSError
+    except OSError as er:
+        print("OSError")
+
+    # Test that closing doesn't raise CancelledError.
+    print("create server4")
+    server4 = await asyncio.start_server(None, "0.0.0.0", 8000)
+    server4.close()
+    await server4.wait_closed()
+
+    async def task(name):
+        print("create", name)
+        srv = await asyncio.start_server(None, "0.0.0.0", 8000)
+        await srv.wait_closed()
+
+    # Test that cancelling the task will still raise CancelledError.
+    t = asyncio.create_task(task("server5"))
+    t.cancel()
+    try:
+        await t
+    except asyncio.CancelledError:
+        print("CancelledError")
+
+    # Test that cancelling the task will still raise CancelledError.
+    t = asyncio.create_task(task("server6"))
+    await asyncio.sleep(0)
```

Comment:
you could combine this and the above test by doing:
```python
for num_sleep in range(2):
    t  = asyncio.create_task(task("server6"))
    for _ in range(num_sleep):
        await asyncio.sleep(0)
    ...
```
and even increase the loop to 3 or 4 (this pattern is used elsewhere to test edge cases of scheduling)